### DISCUSSION TITLE
Is there a way to check the database automatically?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#16302] [Archive]
On complex db we may get some errors very difficult to find.
For instance, I can no more reorder groups and fields in an accordion.
When dragging a new Input field from outside to inside the accordion, or already inside the accordion I get this error

```json
{
  "errors": [
    {
      "message": "Alias type cannot be changed",
      "extensions": {
        "code": "INVALID_PAYLOAD"
      }
    }
  ]
}
```

and after clicking on Dismiss

```json
{
  "errors": [
    {
      "message": "Cannot read properties of undefined (reading 'type')",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

On the other hand, the drag functionality works fine on a new app.

To avoid rebuilding the whole present app, (more than 10 groups of fields within the accordion) I change `sort` and `group` by hand directly on the DB and it works.

They must be some inconsistency in the DB. A check tool would help to see where.

**Comment by azrikahar** [comment]
The error you are facing should be the known issue #16269. It should be fixed by #16270 in patch version `9.20.3` 👍

> They must be some inconsistency in the DB. A check tool would help to see where.

In this particular scenario, the issue is specifically occurring in the API and not the DB, so I'm not sure would any database-centric check tool be able to pinpoint such error.

Generally speaking, if there are any expected changes with any system tables' data (eg. `directus_field` `group` should be a string instead of integer), there will be accompanying migration(s) that should help facilitate said changes during upgrades (with `npx directus database migrate:latest`), so ideally there wouldn't be a need to manually iron out those inconsistencies. 

That being said, the notion of a database check tool sounds interesting nonetheless! Curious to know what do you have in mind on what it should be capable of.

**Reply by Pachat** [reply]
Thank you for the explanation; as of 9.20.4:

- reorder groups and fields in an accordion. 
- dragging a new Input field from outside to inside the accordion 

works like a charm.


The Directus schema is clean and simple. 
When we get some error messages, it is sometime difficult so find if it comes from the code or from some inconsistency in the data.

A database check tool could check:
- do the fields refer to a collection still present
- do the filters (and any other collection, especially the relations) refer to a field still present
- do the permissions refer to a role still present, 
- etc.
